server/src/test/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsTests.java

```diff
@@ -153,12 +157,23 @@ public void testRefreshListeners() throws Exception {
         assertEquals(0, common.refresh.getListeners());
     }
 
+    @SuppressWarnings("unchecked")
+    public void testUuidOnRootStatsIndices() throws IOException {
+        String uuid = createIndex("test").indexUUID();
+        IndicesStatsResponse rsp = client().admin().indices().prepareStats().get();
+        try (XContentParser parser = createParser(JsonXContent.jsonXContent, rsp.toString())) {
```

Comment:
Instead of parsing json, could you use `rsp.getIndex("test")` like the other tests here do and then get the uuid from the `IndexStats` object returned?

Reply:
@rjernst I did that to get coverage on https://github.com/elastic/elasticsearch/pull/31871/files#diff-eaeb35137f4e84d0f731422d07701547R176
If I don't add in the serialization roundtrip I don't get the actual response rendered do I (just checking the return of `rap.getIndex("test")` also passes without that line for serializing the uuid added)?

Reply:
I think testing this directly (not rest) is better as a start. Testing this via rest is a different matter, and could be done through a rest api test, and/or a client test when this is added to the high level rest client.

Reply:
@rjernst thanks for taking a look :)

so, make this test directly for now and leave the REST test for later since that API isn't in the REST client yet?

Reply:
Yes, but you could add this to an existing test under `rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats` to ensure it exists in the rest layer.

Reply:
Done in https://github.com/elastic/elasticsearch/pull/31871/commits/2b932e39b3950cf96d09180ac732ea64976d5282 I think/hope :)